File: src/main/java/split/SplitIndexStream.java
import split.data.LoremIpsum;

void main() {
    var result = splitLoop(LoremIpsum.TEXT);
    println(result);
}

record IndexWord(int index, String value) {
}

IndexWord splitLoop(String line) {
    var pattern = Pattern.compile(" ");
    var words = pattern.split(line);
    return IntStream.range(0, words.length)
            .filter(index -> words[index].length() == 3)
            .mapToObj(index -> new IndexWord(index, words[index]))
            .findFirst()
            .orElseThrow();
}

File: src/main/kotlin/splitkt/SplitIndexStream.kt
package splitkt

import split.data.LoremIpsum

fun main() {
    val result = splitIndexStream(LoremIpsum.TEXT)
    println(result)
}

data class IndexWord(val index: Int, val value: String)

fun splitIndexStream(line: String): IndexWord =
    line.split(" ")
        .withIndex()
        .map { (index, value) -> IndexWord(index, value) }
        .first { it.value.length == 3 }
